### PR TITLE
[fix](cloud tvf) Fix tvf query run in cloud multi cluster

### PR DESCRIPTION
**Pull request by deardeng**
Fix
```
mysql> select * from numbers("number" = "10");
ERROR 1105 (HY000): errCode = 2, detailMessage = There is no scanNode Backend available.[10002: not exist]
```

### CODE REVIEW

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

File: regression-test/suites/cloud_p0/multi_cluster/test_tvf.groovy [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        return;
+    }
+    def options = new ClusterOptions()
+    options.feConfigs += [
+        'cloud_cluster_check_interval_second=1',
+    ]
+    options.cloudMode = true
+
+    def testCase = {
+        for (i = 0; i < 100; i++) {
```
Comment (yujun777) [inline regression-test/suites/cloud_p0/multi_cluster/test_tvf.groovy:32]: def i = 0;

Comment (deardeng) [reply regression-test/suites/cloud_p0/multi_cluster/test_tvf.groovy]: fixed

**Comment by deardeng** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/catalog/StorageVaultMgr.java [hunk 2]
```diff
@@ -176,7 +179,14 @@ public void createHdfsVault(StorageVault vault) throws DdlException {
     }
 
     private void alterSyncVaultTask() {
-        systemInfoService.getAllBackends().forEach(backend -> {
+        List<Backend> bes;
+        try {
+            bes = systemInfoService.getBackendsByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/StorageVaultMgr.java:184]: Need check if backends in current cluster is enough.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/TabletStatMgr.java [hunk 3]
```diff
@@ -51,7 +52,13 @@ public TabletStatMgr() {
 
     @Override
     protected void runAfterCatalogReady() {
-        ImmutableMap<Long, Backend> backends = Env.getCurrentSystemInfo().getIdToBackend();
+        ImmutableMap<Long, Backend> backends;
+        try {
+            backends = Env.getCurrentSystemInfo().getBackendsWithIdByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/TabletStatMgr.java:57]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/clone/TabletScheduler.java [hunk 4]
```diff
@@ -180,7 +180,13 @@ public Rebalancer getRebalancer() {
      * update working slots at the beginning of each round
      */
     private boolean updateWorkingSlots() {
-        ImmutableMap<Long, Backend> backends = infoService.getAllBackendsMap();
+        ImmutableMap<Long, Backend> backends;
+        try {
+            backends = infoService.getBackendsWithIdByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/clone/TabletScheduler.java:185]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/ReplicasProcNode.java [hunk 5]
```diff
@@ -57,8 +58,8 @@ public ReplicasProcNode(long tabletId, List<Replica> replicas) {
     }
 
     @Override
-    public ProcResult fetchResult() {
-        ImmutableMap<Long, Backend> backendMap = Env.getCurrentSystemInfo().getIdToBackend();
+    public ProcResult fetchResult() throws AnalysisException {
+        ImmutableMap<Long, Backend> backendMap = Env.getCurrentSystemInfo().getBackendsWithIdByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/common/proc/ReplicasProcNode.java:62]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/TabletsProcDir.java [hunk 6]
```diff
@@ -65,10 +65,11 @@ public TabletsProcDir(Table table, MaterializedIndex index) {
         this.index = index;
     }
 
-    public List<List<Comparable>> fetchComparableResult(long version, long backendId, Replica.ReplicaState state) {
+    public List<List<Comparable>> fetchComparableResult(long version, long backendId, Replica.ReplicaState state)
+            throws AnalysisException {
         Preconditions.checkNotNull(table);
         Preconditions.checkNotNull(index);
-        ImmutableMap<Long, Backend> backendMap = Env.getCurrentSystemInfo().getIdToBackend();
+        ImmutableMap<Long, Backend> backendMap = Env.getCurrentSystemInfo().getBackendsWithIdByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/common/proc/TabletsProcDir.java:72]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/common/proc/TrashProcDir.java [hunk 7]
```diff
@@ -50,7 +50,13 @@ public class TrashProcDir implements ProcDirInterface {
     private List<Backend> backends = Lists.newArrayList();
 
     public TrashProcDir() {
-        ImmutableMap<Long, Backend> backendsInfo = Env.getCurrentSystemInfo().getIdToBackend();
+        ImmutableMap<Long, Backend> backendsInfo;
+        try {
+            backendsInfo = Env.getCurrentSystemInfo().getBackendsWithIdByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/common/proc/TrashProcDir.java:55]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/common/publish/ClusterStatePublisher.java [hunk 8]
```diff
@@ -62,8 +63,8 @@ public static ClusterStatePublisher getInstance() {
         return INSTANCE;
     }
 
-    public void publish(ClusterStateUpdate state, Listener listener, int timeoutMs) {
-        Collection<Backend> nodesToPublish = clusterInfoService.getIdToBackend().values();
+    public void publish(ClusterStateUpdate state, Listener listener, int timeoutMs) throws UserException {
+        Collection<Backend> nodesToPublish = clusterInfoService.getBackendsWithIdByCurrentCluster().values();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/common/publish/ClusterStatePublisher.java:67]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/common/publish/TopicPublisherThread.java [hunk 9]
```diff
@@ -73,7 +73,13 @@ protected void runAfterCatalogReady() {
         // because it may means workload group/policy is dropped
 
         // step 2: publish topic info to all be
-        Collection<Backend> nodesToPublish = clusterInfoService.getIdToBackend().values();
+        Collection<Backend> nodesToPublish;
+        try {
+            nodesToPublish = clusterInfoService.getAllBackendsWithIdByAllCluster().values();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/common/publish/TopicPublisherThread.java:78]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/common/util/AutoBucketUtils.java [hunk 10]
```diff
@@ -50,7 +57,13 @@ private static int getBENum() {
 
     private static int getBucketsNumByBEDisks() {
         SystemInfoService infoService = Env.getCurrentSystemInfo();
-        ImmutableMap<Long, Backend> backends = infoService.getAllBackendsMap();
+        ImmutableMap<Long, Backend> backends;
+        try {
+            backends = infoService.getBackendsWithIdByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/AutoBucketUtils.java:62]: Need checking too.

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/manager/NodeAction.java [hunk 11]
```diff
@@ -627,7 +627,7 @@ public Object operateBackend(HttpServletRequest request, HttpServletResponse res
             } else if ("DROP".equals(action)) {
                 currentSystemInfo.dropBackends(hostInfos);
             } else if ("DECOMMISSION".equals(action)) {
-                ImmutableMap<Long, Backend> backendsInCluster = currentSystemInfo.getAllBackendsMap();
+                ImmutableMap<Long, Backend> backendsInCluster = currentSystemInfo.getBackendsWithIdByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/manager/NodeAction.java:630]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/restv2/StatisticAction.java [hunk 12]
```diff
@@ -80,7 +81,13 @@ private int getTblCount(Env env) {
 
     private long getDiskOccupancy(SystemInfoService infoService) {
         long diskOccupancy = 0;
-        List<Backend> backends = infoService.getAllBackends();
+        List<Backend> backends;
+        try {
+            backends = infoService.getBackendsByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/restv2/StatisticAction.java:86]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/restv2/StatisticAction.java [hunk 13]
```diff
@@ -89,7 +96,13 @@ private long getDiskOccupancy(SystemInfoService infoService) {
 
     private long getRemainDisk(SystemInfoService infoService) {
         long remainDisk = 0;
-        List<Backend> backends = infoService.getAllBackends();
+        List<Backend> backends;
+        try {
+            backends = infoService.getBackendsByCurrentCluster();
+        } catch (UserException e) {
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/restv2/StatisticAction.java:102]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/load/StreamLoadRecordMgr.java [hunk 14]
```diff
@@ -239,7 +240,13 @@ private void writeUnlock() {
...
     @Override
     protected void runAfterCatalogReady() {
-        ImmutableMap<Long, Backend> backends = Env.getCurrentSystemInfo().getIdToBackend();
+        ImmutableMap<Long, Backend> backends;
+        try {
+            backends = Env.getCurrentSystemInfo().getAllBackendsWithIdByAllCluster();
+        } catch (AnalysisException e) {
+            LOG.warn("Failed to load backends from system info", e);
+            return;
+        }
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/load/StreamLoadRecordMgr.java:249]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/metric/SimpleCoreMetricVisitor.java [hunk 15]
```diff
@@ -128,8 +131,14 @@ public void visitHistogram(String prefix, String name, Histogram histogram) {
     @Override
     public void visitNodeInfo() {
         long feDeadNum = Env.getCurrentEnv().getFrontends(null).stream().filter(f -> !f.isAlive()).count();
-        long beDeadNum = Env.getCurrentSystemInfo().getIdToBackend().values().stream().filter(b -> !b.isAlive())
-                .count();
+        long beDeadNum = 0;
+        try {
+            beDeadNum = Env.getCurrentSystemInfo().getBackendsWithIdByCurrentCluster()
+                    .values().stream().filter(b -> !b.isAlive())
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/metric/SimpleCoreMetricVisitor.java:137]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/policy/PolicyMgr.java [hunk 16]
```diff
@@ -731,7 +731,7 @@ public void alterPolicy(AlterPolicyStmt stmt) throws DdlException, AnalysisExcep
         // log alter
         Env.getCurrentEnv().getEditLog().logAlterStoragePolicy(storagePolicy);
         AgentBatchTask batchTask = new AgentBatchTask();
-        for (long backendId : Env.getCurrentSystemInfo().getIdToBackend().keySet()) {
+        for (long backendId : Env.getCurrentSystemInfo().getBackendsWithIdByCurrentCluster().keySet()) {
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/policy/PolicyMgr.java:734]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/qe/QueryCancelWorker.java [hunk 17]
```diff
@@ -33,7 +38,13 @@ public QueryCancelWorker(SystemInfoService systemInfoService) {
 
     @Override
     protected void runAfterCatalogReady() {
-        List<Backend> allBackends = systemInfoService.getAllBackends();
+        List<Backend> allBackends;
+        try {
+            allBackends = systemInfoService.getAllBackendsWithIdByAllCluster().values().asList();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/qe/QueryCancelWorker.java:43]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 18]
```diff
@@ -2824,7 +2824,7 @@ private void handleShowLastInsert() {
 
     private void handleAdminShowTabletStorageFormat() throws AnalysisException {
         List<List<String>> resultRowSet = Lists.newArrayList();
-        for (Backend be : Env.getCurrentSystemInfo().getIdToBackend().values()) {
+        for (Backend be : Env.getCurrentSystemInfo().getBackendsWithIdByCurrentCluster().values()) {
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ShowExecutor.java:2827]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/qe/cache/CacheCoordinator.java [hunk 19]
```diff
@@ -110,7 +111,7 @@ public void resetBackend() {
         }
         try {
             belock.lock();
-            ImmutableMap<Long, Backend> idToBackend = Env.getCurrentSystemInfo().getIdToBackend();
+            ImmutableMap<Long, Backend> idToBackend = Env.getCurrentSystemInfo().getBackendsWithIdByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/qe/cache/CacheCoordinator.java:114]: Need checking.

File: fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java [hunk 20]
```diff
@@ -3776,7 +3776,7 @@ private TGetBackendMetaResult getBackendMetaImpl(TGetBackendMetaRequest request,
             result.setStatus(new TStatus(TStatusCode.OK));
 
             final SystemInfoService systemInfoService = Env.getCurrentSystemInfo();
-            List<Backend> backends = systemInfoService.getAllBackends();
+            List<Backend> backends = systemInfoService.getBackendsByCurrentCluster();
```
Comment (dataroaring) [inline fe/fe-core/src/main/java/org/apache/doris/service/FrontendServiceImpl.java:3779]: Need checking.

**Review by dataroaring — CHANGES_REQUESTED** [review]

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Comment by deardeng** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**